### ISSUE TITLE
XGAN and Cycle-GAN with improved w-gan

### ISSUE THREAD

**Issue by oiwio** [shaoanlu/faceswap-GAN#12]
Have you tried the two methods?
[XGAN: Unsupervised Image-to-Image Translation for Many-to-Many Mappings](https://arxiv.org/abs/1711.05139)
[Improved Training of Wasserstein GANs](https://arxiv.org/abs/1704.00028)
I tried to use your code, but it's hard to achieve the effect you showed in the "readme".

**Comment by shaoanlu** [comment]
I thought about XGAN and cycleGAN, didn't come up with a good method to combine consistency loss (or cyclic loss) with the current model. Deepfakes' model is based on a denoising autoencoder and introducing consistency loss seems a little weird to me: feeding warped faceA into `model_A2B` to get fake faceB; then feeding this fake faceB into `model_B2A` to get fake faceA for consistency loss (in the case of cycleGAN). 

But cycleGAN is absolutely capable of face swapping, e.g, this well-known [youtube video](https://www.youtube.com/watch?v=Fea4kZq0oFQ) twitted by [Ian Goodfellow](https://twitter.com/goodfellow_ian/status/916360744253382656?lang=en).

Regarding training data, I did not tried any face except the two I showed in readme since, you know, collecting/cleaning data is such a painful task 😂. However, I have a feeling that the target faces are better to have variety. i.e., different source videos, light condition and so on. My target face dataset contains 1k faces from pinterest and other 4k faces from ~5 different videos. 

Besides, its promising to pre-train our model using dataset like [celebA](http://mmlab.ie.cuhk.edu.hk/projects/CelebA.html) that makes model grasp the concept of human face (I did not do this because of limited computing power).

**Comment by oiwio** [comment]
Hello, after using more training data, I become more and more like Daniel Wu, except the chin, so I want to know that can I modify the face_recognition API to capture the face including the chin, instead of only the middle part of the face. Thanks for your help, bro.

**Comment by shaoanlu** [comment]
Since `face_recognition` API returns xy coordinates of detected faces, you can simply expand the xy values so that the bounding boxes covers larger area.

**Comment by RaineyWu** [comment]
for  (佐野ひなこ) which we want to changed the face, how much the amount of the data set? does its amount will affect the performance?

**Comment by RaineyWu** [comment]
> Hello, after using more training data, I become more and more like Daniel Wu, except the chin, so I want to know that can I modify the face_recognition API to capture the face including the chin, instead of only the middle part of the face. Thanks for your help, bro.

hi, oiwio, you mean you increase your face data and train again with Daniel Wu and get the more acceptable output?
